### ISSUE TITLE
Support a specific reader, not just plugin when using Viewer.open()

### ISSUE THREAD

**Issue by adamltyson** [napari/napari#2432]
From [zulip](https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Specify.20specific.20reader.20within.20a.20plugin/near/230021440) cc @tlambert03 
## Motivation

I have a number of different readers bundled within a single plugin, but I often want to specify one of these. 

## Pitch

To be able to specify a specific reader as well as a plugin. E.g.:
```python
viewer.open(data, plugin=plugin, reader=reader)
```

## Alternatives

I make each reader a separate plugin, but this complicates the list of plugins.

**Comment by sofroniewn** [comment]
Seems like a very reasonable request. I could imagine it coming up in other plugin contexts too, like writers, or analysis etc, so I wonder if we'll have a more general `kwarg` than `reader` that could be an approach we'd reuse in other contexts, i.e. you specify the `plugin=....` and then ?

Maybe plugin could become a tuple optionally `plugin = (plugin_name, my_function_name)`? I like that because it keeps the plugin information together in one place. Curious what others think about the API?

**Comment by tlambert03** [comment]
> plugin = (plugin_name, my_function_name)

I kinda like that.  @jni?  @adamltyson?

**Comment by adamltyson** [comment]
In an ideal world, I could specify a reader name, without the plugin name. However, I wouldn't be using this part of the API on a regular basis, only within other analysis plugins.

Not specifying the plugin name could lead to issues with non-unique names, but the existing hierarchy in the plugin sorter could take care of this. 

TLDR: If it's possible, I don't mind how.

**Comment by sofroniewn** [comment]
I think we'll probably need a plugin name so avoid potential name space conflicts. I guess we could just use the hierarchy in the sorter to find the first reader name that matches, but having both seems like the safer place to start. Thanks for understanding.

**Comment by tlambert03** [comment]
yeah... I'm afraid I agree here. too easy to run into namespace conflicts

**Comment by adamltyson** [comment]
Sounds good.
